### PR TITLE
Add some more integral functions

### PR DESCRIPTION
**Pull request by bnjones**
Still to do:
- [x] Extend `_MPMathFunction` to support dispatching to multiple mpmath functions
- [x] Use mpmath to implement the two-argument form of `Erf`
- [x] Handle out-of-range numeric arguments to `InverseErf`

### CODE REVIEW

**Comment by sn6uv** [comment]
Looks very similar to a77c08e66e6103d28af3c09e69e0d857600c10da.

File: mathics/builtin/specialfunctions.py [hunk 1]
```diff
@@ -34,9 +36,64 @@ class Erf(_MPMathFunction):
...
+    <dt>'InverseErf[$z$]'
+        <dd>returns the inverse error function of $z$.
+    </dl>
+
+    >> Erf[1.0]
+     = 0.842700792949714869
+    >> Erf[0]
+     = 0
+    >> Plot[Erf[x], {x, -2, 2}]
+     = -Graphics-
```
Comment (sn6uv) [inline mathics/builtin/specialfunctions.py]: These should be `InverseErf` not `Erf`.

File: mathics/builtin/specialfunctions.py [hunk 2]
```diff
@@ -34,9 +36,64 @@ class Erf(_MPMathFunction):
...
+
+class InverseErfc(SympyFunction):
+    """
+    <dl>
+    <dt>'InverseErfc[$z$]'
+        <dd>returns the inverse complementary error function of $z$.
+    </dl>
+
+    >> Erf[1.0]
+     = 0.842700792949714869
```
Comment (sn6uv) [inline mathics/builtin/specialfunctions.py]: `Should be InverseErfc`

**Comment by bnjones** [comment]
Happy to rebase this on a77c08e if you like? They're pretty much the same but there are some tests in that commit that I think it'd be good to have.

**Comment by sn6uv** [comment]
I think it's fine not to rebase. Feel free to copy the tests if you think they're useful. I am working on fixing some precision issues related to `_MPMathFunction` (but I don't there are actually any changes to that class), see #440.

**Comment by sn6uv** [comment]
Looks good :+1: , anything more to add?

**Comment by bnjones** [comment]
Great, thanks for the review! Still got a few (small) things to add, I'll push these shortly.

**Comment by sn6uv** [comment]
Looks good, shall I merge this?

**Comment by bnjones** [comment]
Yep, if you're happy with it I'm happy with it

**Pull request merged**